File: InterFatecs/2024/Primeira fase/calcetis/calcetis.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main(void){
    int v, n, p, max;
    cin >> v >> n;
    vector<int> prod;
    if (v <= 110){
        max = 140-v;/* 200-60-v resulta no max possível, 140 = 200-60. Uso 60 porque são os dois valores mínimos 30,
        possiveis*/
        while (n != 0){
            n--;
            cin >> p;
            if(p <=  max){
                prod.push_back(p);
            }
        }
        n = prod.size();
        
        for (int i = 0; i < n-2; i++){
            for (int j = i+1 ; j < n-1; j++){
                for (int k = j+1; k < n; k++){
                    if (v + prod[i] + prod[j] + prod[k] == 200){
                        cout << "fretegratis" << "\n";
                        return 0; 
                    }
                }
                
            }
            
        }
        
    }
    cout << "fretepago\n"; 
    return 0; 
}

File: InterFatecs/2024/Primeira fase/calcetis/calcetis.py
v, n = [int(i) for i in input().split()]
x = []
while n != 0:
    z = int(input())
    if z >= 110:
        continue
    x.append(z)
    n -= 1
resp = "fretepago"

c = len(x)
if c%2 == 0:
    c2 = c//2
else:
    c2 = c//2 + 1
for i in range(c2):
    for j in range(i+1,c):
        for k in range(j+1,c):
            if v + x[i] + x[j] + x[k] == 200:
                resp = "fretegratis"
                break
print(resp)
